app/templating/summary/question.py

```diff
@@ -17,17 +17,15 @@ def __init__(self, question_schema, answers):
         self.number = question_schema.get('number', None)
         self.answers = self._build_answers(question_schema, answer_schema, answers)
 
-    def is_skipped(self, all_answers):
+    def is_skipped(self, answer_store, metadata):
         if self.skip_condition is not None:
-            for when_rule in self.skip_condition['when']:
-                answer = all_answers.get(when_rule['id'])
-                if isinstance(answer, Markup):
-                    answer = answer.unescape()
-
-                if not evaluate_rule(when_rule, answer):
-                    return False
-            return True
-
+            for when_rules in self.skip_condition:
+                for when_rule in when_rules['when']:
+                    if 'id' in when_rule:
+                        answer = answer_store.map().get(when_rule['id'])
```

Comment:
We shouldn't use the map method to get an answer, we should be using the `filter` method on `answer_store` to return all matching answers.